rts/Rendering/Textures/Bitmap.cpp

```diff
@@ -630,74 +630,95 @@ void TBitmapAction<T, ch>::Renormalize(const float3& newCol)
+	// below.
 	CBitmap tmp(nullptr, bmp->xsize, bmp->ysize, bmp->channels, bmp->dataType);
+	CBitmap tmp2(nullptr, bmp->xsize, bmp->ysize, bmp->channels, bmp->dataType);
 
-	CBitmap* src =  bmp;
-	CBitmap* dst = &tmp;
-
-	//don't use "this" here
-	auto srcAction = BitmapAction::GetBitmapAction(src);
-	auto dstAction = BitmapAction::GetBitmapAction(dst);
+	std::array<CBitmap*, 3> bitmaps = {bmp, &tmp, &tmp2};
+	std::array<std::unique_ptr<BitmapAction>, 3> actions = {
+		BitmapAction::GetBitmapAction(bitmaps[0]),
+		BitmapAction::GetBitmapAction(bitmaps[1]),
+		BitmapAction::GetBitmapAction(bitmaps[2])
+	};
 
 	using ThisType = decltype(this);
 
 	for (int iter = 0; iter < iterations; ++iter) {
-		for_mt(0, src->ysize, [&](const int y) {
-			for (int x = 0; x < src->xsize; x++) {
-				int yBaseOffset = (y * src->xsize);
-				for (int a = 0; a < src->channels; a++) {
+		for(int dimension = 0; dimension < 2; ++dimension) {
 
-					///////////////////////////////////////
-					float fragment = 0.0f;
+			CBitmap* src = bitmaps[dimension];
+			CBitmap* dst = bitmaps[dimension + 1];
 
-					for (int i = 0; i < 9; ++i) {
-						int yoffset = (i / 3) - 1;
-						int xoffset = (i - (yoffset + 1) * 3) - 1;
+			auto& srcAction = actions[dimension];
+			auto& dstAction = actions[dimension + 1];
 
-						const int tx = x + xoffset;
-						const int ty = y + yoffset;
+			for_mt(0, src->ysize, [&](const int y) {
+				for (int x = 0; x < src->xsize; x++) {
+					int yBaseOffset = (y * src->xsize);
+					for (int a = 0; a < src->channels; a++) {
+						float fragment = 0.0f;
 
-						xoffset *= ((tx >= 0) && (tx < src->xsize));
-						yoffset *= ((ty >= 0) && (ty < src->ysize));
+						for (int i = 0; i < 3; ++i) {
+							int yoffset = dimension == 1 ? (i - 1) : 0;
+							int xoffset = dimension == 0 ? (i - 1) : 0;
 
-						const int offset = (yoffset * src->xsize + xoffset);
+							const int tx = x + xoffset;
+							const int ty = y + yoffset;
 
-						auto& srcChannel = static_cast<ThisType>(srcAction.get())->GetRef(yBaseOffset + x + offset, a);
+							xoffset *= ((tx >= 0) && (tx < src->xsize));
+							yoffset *= ((ty >= 0) && (ty < src->ysize));
 
-						const float thisWeight = mix(1.0f, weight, i == 4);
-						fragment += (thisWeight * blurkernel[i] * srcChannel);
-					}
+							const int offset = (yoffset * src->xsize + xoffset);
 
-					auto& dstChannel = static_cast<ThisType>(dstAction.get())->GetRef(yBaseOffset + x, a);
+							auto& srcChannel = static_cast<ThisType>(srcAction.get())->GetRef(yBaseOffset + x + offset, a);
 
-					if constexpr (std::is_same_v<ChanType, float>) {
-						dstChannel = static_cast<ChanType>(std::max(fragment, 0.0f));
-					}
-					else {
-						dstChannel = static_cast<ChanType>(std::clamp(fragment, 0.0f, static_cast<float>(GetMaxNormValue())));
+							fragment += (blurkernel[i] * srcChannel);
+						}
+
+						// The original blur algorithm is not a gaussian blur, as it weights
```

Comment:
Does "the original algorithm" refer to the one your commit is replacing? If so then people reading the comment will not know what it refers to (or if they do, what is its significance) because it is no longer there. This sort of description comparing to the old implementation belongs in the commit message